src/main/java/com/programmers/application/CLIVoucherApplicationRunner.java

```diff
@@ -0,0 +1,26 @@
+package com.programmers.application;
+
+import com.programmers.application.controller.Controller;
+import com.programmers.application.controller.FrontController;
+import org.springframework.boot.ApplicationArguments;
+import org.springframework.boot.ApplicationRunner;
+import org.springframework.context.ApplicationContext;
+import org.springframework.stereotype.Component;
+
+@Component
+public class CLIVoucherApplicationRunner implements ApplicationRunner {
+    private final ApplicationContext applicationContext;
+
+    public CLIVoucherApplicationRunner(ApplicationContext applicationContext) {
+        this.applicationContext = applicationContext;
+    }
+
+    @Override
+    public void run(ApplicationArguments args) throws Exception {
+        applicationContext.getBean(Controller.class);
+        FrontController controller = applicationContext.getBean(FrontController.class);
+        while (true) {
+            controller.process();
+        }
+    }
```

Comment:
```suggestion
    private final FrontController frontController;

    public CLIVoucherApplicationRunner(FrontController frontController) {
        this.frontController = frontController;
    }

    @Override
    public void run(ApplicationArguments args) throws Exception {
        while (true) {
            frontController.process();
        }
    }
```

무겁게 애플리케이션 컨텍스트를 들고있을 필요 없이, FrontController만 주입받으면 될 것 같아요.